src/System.Diagnostics.EventLog/tests/EventLogTests/EventLogWriteEntryTests.cs

```diff
@@ -394,7 +482,21 @@ internal static class EventLogEntryCollectionExtensions
     {
         internal static EventLogEntry LastOrDefault(this EventLogEntryCollection elec)
         {
-            return elec.Count > 0 ? elec[elec.Count - 1] : null;
+            bool entryRetrieved = true;
+            EventLogEntry eventLogEntry = null;
+            while (entryRetrieved)
+            {
+                try
+                {
+                    eventLogEntry = elec.Count > 0 ? elec[elec.Count - 1] : null;
+                    entryRetrieved = false;
+                }
+                catch (Win32Exception)
```

Comment:
All your loops should have retry counts so they don't hang for 20 minutes (the global timeout) but instead no more than they need to (eg typical delay x 3 or suchlike).